drivers/input/input_sbus.c

```diff
@@ -0,0 +1,363 @@
+			data->xfer_bytes +=
+				uart_fifo_read(uart_dev, &data->rd_data[data->xfer_bytes],
+					       SBUS_FRAME_LEN - data->xfer_bytes);
+			if (data->xfer_bytes == SBUS_FRAME_LEN) {
+				/* Transfer took longer then 4ms probably faulty */
+				if (k_uptime_get_32() - data->last_rx_time > 4) {
+					data->xfer_bytes = 0;
+					data->partial_sync = false;
+				} else if (data->rd_data[0] == SBUS_HEADER &&
+					   data->rd_data[SBUS_FRAME_LEN - 1] == SBUS_FOOTER) {
+					data->in_sync = true;
+				} else {
+					/* Dummy read to clear fifo */
+					uart_fifo_read(uart_dev, &data->rd_data[0], 1);
+					data->xfer_bytes = 0;
+					data->partial_sync = false;
+				}
+			}
+		} else {
+			if (uart_fifo_read(uart_dev, &data->rd_data[0], 1) == 1) {
+				if (data->rd_data[0] == SBUS_HEADER) {
+					data->partial_sync = true;
+					data->xfer_bytes = 1;
+					data->last_rx_time = k_uptime_get_32();
+				}
+			}
+		}
+	}
+
+	if (data->in_sync && (k_uptime_get_32() - data->last_rx_time > SBUS_INTERFRAME_SPACING)) {
+		data->partial_sync = false;
+		data->in_sync = false;
+		data->xfer_bytes = 0;
+		k_sem_give(&data->report_lock);
+	} else if (data->in_sync && data->xfer_bytes == SBUS_FRAME_LEN) {
+		data->xfer_bytes = 0;
+
+		if (data->rd_data[0] == SBUS_HEADER &&
+		    data->rd_data[SBUS_FRAME_LEN - 1] == SBUS_FOOTER) {
+			memcpy(data->sbus_frame, data->rd_data, SBUS_FRAME_LEN);
+			k_sem_give(&data->report_lock);
+		} else {
+			data->partial_sync = false;
+			data->in_sync = false;
+		}
+	}
+}
+
+/*
+ * @brief Initialize sbus driver
+ */
+static int input_sbus_init(const struct device *dev)
+{
+	const struct input_sbus_config *const config = dev->config;
+	struct input_sbus_data *const data = dev->data;
+	int i, ret;
+
+	uart_irq_rx_disable(config->uart_dev);
+	uart_irq_tx_disable(config->uart_dev);
+
+	LOG_DBG("Initializing SBUS driver");
+
+	for (i = 0; i < SBUS_CHANNEL_COUNT; i++) {
+		data->last_reported_value[i] = 0;
+		data->channel_mapping[i] = -1;
+	}
+
+	data->xfer_bytes = 0;
+	data->in_sync = false;
+	data->partial_sync = false;
+	data->last_rx_time = 0;
+
+	for (i = 0; i < config->num_channels; i++) {
+		data->channel_mapping[config->channel_info[i].sbus_channel - 1] = i;
+	}
+
+	ret = uart_configure(config->uart_dev, &uart_cfg_sbus);
+	if (ret < 0) {
+		LOG_ERR("Unable to configure UART port: %d", ret);
+		return ret;
+	}
+
+	ret = uart_irq_callback_user_data_set(config->uart_dev, config->cb, (void *)dev);
+	if (ret < 0) {
+		if (ret == -ENOTSUP) {
+			LOG_ERR("Interrupt-driven UART API support not enabled");
+		} else if (ret == -ENOSYS) {
+			LOG_ERR("UART device does not support interrupt-driven API");
+		} else {
+			LOG_ERR("Error setting UART callback: %d", ret);
+		}
+		return ret;
+	}
+
+	uart_irq_rx_enable(config->uart_dev);
+
+	/* Initialize semaphore used by thread to report input */
+	k_sem_init(&data->report_lock, 0, 1);
+
+	k_thread_create(&data->thread, data->thread_stack, CONFIG_INPUT_SBUS_THREAD_STACK_SIZE,
```

Comment:
```suggestion
	k_thread_create(&data->thread, data->thread_stack, K_KERNEL_STACK_SIZEOF(data->thread_stack),
```

I think that this may be different in some architecture that have reserved space